Fix: Mobile friendlyness

danielsolistensvik opened the pull request:
## Features:
* Fixed showing of NFT's on account detail page. 
* Fixed appname in appbar to be hidden on small devices. 
* Fixed card sizes and breakpoints on overview page. 

## View
<img src="https://user-images.githubusercontent.com/58626043/165085695-4e82ddcf-9cd3-44e2-8d38-44e2472a47c1.png" data-canonical-src="https://user-images.githubusercontent.com/58626043/165085695-4e82ddcf-9cd3-44e2-8d38-44e2472a47c1.png" width="30%" height="30%" />  <img src="https://user-images.githubusercontent.com/58626043/165086294-6d1fc106-e64b-4e55-9ceb-1e8277029c52.png" data-canonical-src="https://user-images.githubusercontent.com/58626043/165086294-6d1fc106-e64b-4e55-9ceb-1e8277029c52.png" width="30%" height="30%" />

danielsolistensvik commented:
Only mobile friendlyness in this PR :)

modersohn commented:
Thanks for resubmitting this more isolated! 

But somehow a couple of older master commits got dragged into your PR - that's why we're also seeing older comments from me and fudgey here. 

I suspect this is because your fix-branch did not originate from an up-to date dev-branch, but from an outdated master-branch. 

Hopefully this will just resolve when this is merged, but in the future, please make sure that your dev-branch is up to date and then start your PR-branch from there.

I'll try to get around reviewing this tonight - if fudgey doesn't beat me to it ;-)

fudgebucket27 commented:
It does look a lot better on mobile now! But yes we need to find a balance between making it more mobile friendly and desktop usage.

danielsolistensvik commented:
Super!
1. Fixed comes down to screen size. It's snaps to the "correct" breakpoint automatically based ojn your screen size.
Have added padding left-right to 0, so that it fills the width.
2. Have now set xs=12, sm and md 4, lg 2.
3. Fixed it with positioning 

Seems to be better now both in large and small displays.

modersohn commented:
Yay, you're getting there!

Regarding number 3: wow, that is what I always wanted to achieve, but couldn't, and you made it! Excellent!

But (isn't there always one): while the page loads, it does indeed now use the whole screen width - but then something makes it snap back again?! So while we now have 2 rows with 6 NFTs again, it still does look a bit skewed:

<img width="1679" alt="image" src="https://user-images.githubusercontent.com/44807458/165239520-3d7e8894-7479-4f42-9588-c695925c81ce.png">

Can you find a way to fix this?

danielsolistensvik commented:
Sweet :)  
I'm trying to recreate the issue you're having, but I can't seem to get the same issue in my browsers.  
In both Chrome and Brave it's always stretched and never snaps back in with margins like yours. 
Even tried Edge now, and it never snaps back. 
Can you try to run it again and hard-refresh so no css is in cache or anything.

modersohn commented:
Oh, that's surprising. I'm using Safari on macOS - tried hard refresh, private browsing window all to no avail. It still behaves like that. That's odd.

danielsolistensvik commented:
One question. Do you have a really wide screen? 😄 Maybe that's why? 
I have added MaxWidth now to extraextralarge. Check again to see if that helps

modersohn commented:
Yes that's a 4K monitor and no the ExtraExtraLarge did not really improve things:

<img width="1901" alt="image" src="https://user-images.githubusercontent.com/44807458/165316900-fdd282ad-40cd-47e6-951d-b4557bea3d7e.png">

So it seems we end up with a MaxWidth of 1280px and that is, well, not wide enough ;-) But seriously, why do we have to set MaxWidth at all? Or is that just how MudBlazor and it's breakpoints work? Would it be okay if we hardcode something bigger? 1800px looks nice on my machine ;-)

danielsolistensvik commented:
Hmm, I'm not one of the lucky ones having a 4k monitor then 😄 
I think maybe instead of hardcoding it, maybe leaving MaxWidth unset is the way to go. It's set to the container in MainLayout and also to the Account container. If we leave it out completely, maybe it will be set to the correct size on all displays. I'll give it a try if it still works on my screen and mobil. And then commit so you can try it out on yours.

danielsolistensvik commented:
Nope, when removing it was not working as expected on either desktop or mobile. And if MaxWidth set to False, then it is also not working as expected for mobile. This is tricky... Tried to see if there was an issue open for MaxWidth and bigger screens, but there was not :/  
Maybe hardcoding is the way to go for now?

modersohn commented:
Hm, is there no way to set MaxWidth to false for large only? I presume not, I'm sure you would have tried.

So if you see no other way, yes, I'd say hardcode MaxWidth to 1800px with a comment explaining that - at least for now - even using MaxWidth.ExtraExtraLarge you end up with 1280px - which is not enough.

I'm surprised this hasn't been reported to MudBlazor - it is something that they actually need to improve, right? 

This is how [this snipped](https://try.mudblazor.com/snippet/mYwGYocgzpKqqVZK) looks like on my screen:
<img width="1910" alt="image" src="https://user-images.githubusercontent.com/44807458/165336775-ced034a7-7ac8-4057-a779-76019e40df48.png">

Do you think this is sufficient info for opening an issue with them?

fudgebucket27 commented:
Are we on the latest version of MudBlazor? Maybe there is a fix in a later
version?

On Wed, 27 Apr 2022, 1:29 am modersohn, ***@***.***> wrote:

> Hm, is there no way to set MaxWidth to false for large only? I presume
> not, I'm sure you would have tried.
>
> So if you see no other way, yes, I'd say hardcode MaxWidth to 1800px with
> a comment explaining that - at least for now - even using
> MaxWidth.ExtraExtraLarge you end up with 1280px - which is not enough.
>
> I'm surprised this hasn't been reported to MudBlazor - it is something
> that they actually need to improve, right?
>
> This is how this snipped
> <https://try.mudblazor.com/snippet/mYwGYocgzpKqqVZK> looks like on my
> screen:
> [image: image]
> <https://user-images.githubusercontent.com/44807458/165336775-ced034a7-7ac8-4057-a779-76019e40df48.png>
>
> Do you think this is sufficient info for opening an issue with them?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/fudgebucket27/Lexplorer/pull/115#issuecomment-1109940734>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABIDWT6UOAXJX2TRX4T7P4DVHADWBANCNFSM5UH6WCKQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

danielsolistensvik commented:
I have made a new app.css which has a class called extra-extra-extra-large 😄  
I'm now using this in the containers and it seems to be working on all devices. Largest is now set to 1980px.  
Also added for even smaller devices a new maxwidth.  
From all I could find on this it seems to be a standard or 1200-> has all the same settings. So this may be why it's not added in MudBlazor.  
Try it out and see if it fits your needs :)  
I have also made the MudNumericField a bit smaller so it wouldn't stretch the container on smaller devices. It still looks great on large device, at least here. But have a look at yours and see if it looks OK.

modersohn commented:
Wow, it looks gorgeous now!

1. I've noticed you only made the page edit for transactions smaller - I presume because it has less space because of the "Transactions" heading next to it. It looks a little bit odd on the desktop where you see both page number edits and one is a lot smaller. I mean it's a small glitch only and I can certainly live with it.
2. If I enter "Responsive design" mode on Safari and watch the same page with a simulated iPad Pro it looks like this:
<img width="1166" alt="image" src="https://user-images.githubusercontent.com/44807458/165349534-8871fec5-295e-4de5-b3ce-11c9dffdd4c8.png">
Is there any way to make the background pictures more square? It only happens with that specific device and again, this is a very minor issue - just in case you have an idea.

Besides these 2 minor things, I'm all for merging this!

fudgebucket27 commented:
Excllent. I approve the merge too. Can you merge this into Dev then make a
PR nd merge into prod @modersohn?

On Wed, 27 Apr 2022, 2:38 am modersohn, ***@***.***> wrote:

> Wow, it looks gorgeous now!
>
>    1. I've noticed you only made the page edit for transactions smaller -
>    I presume because it has less space because of the "Transactions" heading
>    next to it. It looks a little bit odd on the desktop where you see both
>    page number edits and one is a lot smaller. I mean it's a small glitch only
>    and I can certainly live with it.
>    2. If I enter "Responsive design" mode on Safari and watch the same
>    page with a simulated iPad Pro it looks like this:
>
> [image: image]
> <https://user-images.githubusercontent.com/44807458/165349534-8871fec5-295e-4de5-b3ce-11c9dffdd4c8.png>
> Is there any way to make the background pictures more square? It only
> happens with that specific device and again, this is a very minor issue -
> just in case you have an idea.
>
> Besides these 2 minor things, I'm all for merging this!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/fudgebucket27/Lexplorer/pull/115#issuecomment-1110012474>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABIDWT3TB74424WXTWSK6BDVHALXHANCNFSM5UH6WCKQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

modersohn commented:
OK will do!

@danielsolistensvik just in case you have solutions for the remaining minor issues, I'll gladly look at another PR ;-) 

But for now a huge thank you for this one already!

danielsolistensvik commented:
Sorry, I've not been on my computer since my last comment. Only happy to help with this super project :) 
Will push more if I find more to add at a later point, or if it is anything you want me to do just give me a nod.
